### ISSUE TITLE
`distributed.tests.test_client.test_nested_compute` failing

### ISSUE THREAD

**Issue by fjetter** [dask/distributed#7483]
https://github.com/dask/distributed/actions/runs/3932865717/jobs/6725937308

`distributed.tests.test_client.test_nested_compute` seems to pretty consistently fail on win-py3.8

![image](https://user-images.githubusercontent.com/8629629/212933040-1d48d6bf-6139-41e7-a511-07898a4890e4.png)

I'm surprised this is only happening on windows. I can reproduce this locally as well

```python
2023-01-16 18:13:03,441 - distributed.worker - WARNING - Compute Failed

Key:       fib-10a1838a5b9ea81099d4d0a02946bb70

Function:  fib

args:      (8)

kwargs:    {}

Exception: "RuntimeError('Requested dask.distributed scheduler but no Client active.')"
```

I suspect this is a regression that was introduced by https://github.com/dask/dask/pull/9808 (although I don't understand how)

**Comment by hendrikmakait** [comment]
Re-opened by #7803 which reverts the fix from #7482

**Comment by hendrikmakait** [comment]
Looking at the test report, this seems to be resolved.
